src/components/modal/z-modal/styles.css

```diff
@@ -137,16 +148,24 @@
     position: fixed;
     top: 0;
     left: 0;
-    width: 100%;
-    height: 100%;
+    overflow: hidden;
+    width: 100vw;
+    height: 100vh;
     background-color: var(--gray900);
     opacity: 0.7;
   }
 
+  .modal-background-variant {
```

Comment:
In base a quanto sopra, questa classe la potresti usare solo per modificare il comportamente del position